File: 프로그래머스/2/77485. 행렬 테두리 회전하기/행렬 테두리 회전하기.java
class Solution {
    public int[] solution(int rows, int columns, int[][] queries) {
        int[] answer = new int[queries.length];

        int[][] arr = new int[rows+1][columns+1];

        int num = 1;
        for(int i = 1 ; i <= rows; i++) {
            for(int j = 1;  j <= columns; j++) {
                arr[i][j] = num++;
            }
        }

        for(int i = 0 ; i < queries.length; i++) {
            int x1 = queries[i][0];
            int y1 = queries[i][1];
            int x2 = queries[i][2];
            int y2 = queries[i][3];

            int temp = arr[x1][y1];

            int min = temp;

            //좌측으로 이동
            for(int j = x1; j < x2; j++) {
                arr[j][y1] = arr[j+1][y1];
                min = Math.min(min, arr[j][y1]);
            }

            //하단으로 이동
            for(int j = y1; j < y2; j++) {
                arr[x2][j] = arr[x2][j+1];
                min = Math.min(min, arr[x2][j]);
            }

            //우측으로 이동
            for(int j = x2; j > x1; j--) {
                arr[j][y2] = arr[j-1][y2];
                min = Math.min(min, arr[j][y2]);
            }


            //상단으로 이동
            for(int j = y2; j > y1; j--) {
                arr[x1][j] = arr[x1][j-1];
                min = Math.min(min, arr[x1][j]);
            }

            arr[x1][y1+1] = temp;

            answer[i] = min;
        }

        return answer;
    }
}

File: 프로그래머스/2/77485. 행렬 테두리 회전하기/행렬 테두리 회전하기.kt
import kotlin.math.min

class Solution {
    fun solution(rows: Int, columns: Int, queries: Array<IntArray>): IntArray {
        var answer = intArrayOf()
        
        var arr = Array(rows+1, {Array(columns+1, {0})})
        
        var num = 1
        for(i in 1 .. rows){
            for(j in 1 .. columns){
                arr[i][j] = num++
            }
        }
        
        for((index, element) in queries.withIndex()){
            var x1 = element[0]
            var y1 = element[1]
            var x2 = element[2]
            var y2 = element[3]
            
            var temp = arr[x1][y1]
            
            var min = temp
            
            //아래에서 위로 이동
            for(i in x1 until x2){
                arr[i][y1] = arr[i+1][y1]
                min = min(min, arr[i][y1])
            }
            //arr[2][2] = arr[3][2] = 14
            //arr[3][2] = arr[4][2] = 20
            //arr[4][2] = arr[5][2] = 26
            
            //오른쪽에서 왼쪽으로 이동
            for(i in y1 until y2) {
                arr[x2][i] = arr[x2][i+1]
                min = min(min, arr[x2][i])
            }
            //arr[4][2] = arr[4][3] = 27
            //arr[4][3] = arr[4][4] = 28
            
            //위에서 아래로 이동
            for(i in x2 downTo x1+1) {
                arr[i][y2] = arr[i-1][y2]
                min = min(min, arr[i][y2])
            }
            //arr[5][4] = arr[4][4] = 22
            //arr[4][4] = arr[3][4] = 16
            //arr[3][4] = arr[2][4] = 10

            
            //왼쪽에서 오른쪽으로 이동
            for(i in y2 downTo y1+1) {
                arr[x1][i] = arr[x1][i-1]
                min = min(min, arr[x1][i])
            }
            //arr[2][4] = arr[2][3] = 9
            //arr[2][3] = arr[2][2] = 14
            
            //맨 처음 start 끊은 값으로 바꿔주기
            arr[x1][y1+1] = temp

            //최솟값 저장
            answer += min
            
        }
        return answer
    }
}